pkg/provisioner/ironic/ironic.go

```diff
@@ -1784,6 +1790,14 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 	return result, nil
 }
 
+func (p *ironicProvisioner) hostName() string {
+	if !p.watchAllNamespaces || p.host.Namespace == p.podNamespace {
+		return p.host.Name
+	}
+
+	return p.host.Namespace + "-" + p.host.Name
```

Comment:
ok, so would the "." be better? I thought it's a "hostname" and a "." might be seen as a sub-domain - but maybe this is not an issue.

Reply:
What about `/`? Maybe ironic doesn't allow that?

Reply:
I think we expect it to be a valid hostname, I don't think we allow `/`

Reply:
@dtantsur what character would you suggest?

Reply:
Note that we pass the BMH (not Ironic) name as metadata to the host:

https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L1399

So although we'll need to align with the ironic validation constraints, I don't think this name actually ever gets used as a hostname on the target OS?

Reply:
I wonder about using a UUID for the hostname?  Sounds like we don't use it for lookup so it shouldn't affect that.  This would mess with debuggability a bit but could make this all a lot easier.

Reply:
> I wonder about using a GUID for the hostname? Sounds like we don't use it for lookup so it shouldn't effect that. This would mess with debuggability a bit but could make this all a lot easier.

Personally I think it would affect debuggability a lot, as a developer there's a lot of value in being able to interact with the Ironic API directly and be able to see at a glance which hosts align with specific BMH resources.

Reply:
Yeah good point.